lightning_client.go

```diff
@@ -2703,3 +2708,60 @@ func (s *lightningClient) NetworkInfo(ctx context.Context) (*NetworkInfo,
 		NumZombieChans:       resp.NumZombieChans,
 	}, nil
 }
+
+// InvoiceSubscription holds the parameters to specify from which update
+// to start streaming
+type InvoiceSubscription struct {
+	//
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all added indexes with an add_index greater than this
+	//value. This allows callers to catch up on any events they missed while they
+	//weren't connected to the streaming RPC.
+	AddIndex uint64
+	//
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all settled indexes with an settle_index greater than
+	//this value. This allows callers to catch up on any events they missed while
+	//they weren't connected to the streaming RPC.
+	SettleIndex uint64
+}
+
+// SubscribeInvoices allows a client to subscribe to updates
+// of newly added/settled invoices.
+func (s *lightningClient) SubscribeInvoices(ctx context.Context, req InvoiceSubscription) (<-chan *lnrpc.Invoice, <-chan error, error) {
+	rpcCtx, cancel := context.WithTimeout(ctx, rpcTimeout)
+	defer cancel()
+
+	rpcCtx = s.adminMac.WithMacaroonAuth(rpcCtx)
+	invoicesStream, err := s.client.SubscribeInvoices(rpcCtx, &lnrpc.InvoiceSubscription{
+		AddIndex:    req.AddIndex,
+		SettleIndex: req.SettleIndex,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	invoicesUpdates := make(chan *lnrpc.Invoice)
+	streamErr := make(chan error, 1)
+
+	// Backups updates goroutine.
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		for {
+			invoices, err := invoicesStream.Recv()
```

Comment:
nit: it's just one invoice, variable can be in singular.

Reply:
I was thinking of the plurality of the htlc in the invoices. I've fixed it.